extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/HttpConfig.java

```diff
@@ -0,0 +1,51 @@
+package io.quarkus.vertx.http.runtime;
+
+import static io.quarkus.runtime.annotations.ConfigPhase.RUN_TIME;
+
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigDocIgnore;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+import io.smallrye.config.WithParentName;
+
+/**
+ * Partial mapping of <code>quarkus.http</code> to help with the assignment of random ports in
+ * {@link VertxConfigBuilder}. Can be removed and replaced when {@link HttpConfiguration} moves to
+ * {@link ConfigMapping}.
+ */
+@ConfigMapping(prefix = "quarkus.http")
+@ConfigRoot(phase = RUN_TIME)
+public interface HttpConfig {
+    @ConfigDocIgnore
+    @WithDefault("8080")
+    int port();
+
+    @ConfigDocIgnore
+    @WithDefault("8081")
+    int testPort();
+
+    @ConfigDocIgnore
+    String host();
+
+    @ConfigDocIgnore
+    Optional<String> testHost();
+
+    @ConfigDocIgnore
+    @WithDefault("true")
+    boolean hostEnabled();
+
+    @ConfigDocIgnore
+    @WithDefault("8443")
+    int sslPort();
+
+    @ConfigDocIgnore
+    @WithDefault("8444")
+    int testSslPort();
+
+    @ConfigDocIgnore
```

Comment:
Why no doc?

Reply:
The new mappings that I've added are temporary and incomplete. These are to map the configuration we need to reference in the `ConfigSourceFactory`. Unfortunately, the old roots cannot do this, and I also didn't want to mix root -> mapping changes in this PR.

I've added javadocs on each of them. Example:
https://github.com/quarkusio/quarkus/pull/43220/files#diff-28531acf5fb5dfd40a5b24cf9a79a367c55bdf41f10ce4608bb189d53e78c8b7R14-R18

Reply:
The no doc is to not clash with the main root configs for each case, which still exist and are used across the extension. These new mappings are only for the context of the factory. Once we move to `@ConfigMapping` we can drop these and reference the new ones.